src/main/java/uk/ac/ebi/eva/pipeline/io/writers/VepProcess.java

```diff
@@ -118,21 +137,28 @@ private void captureOutput(Process process, String vepOutput) {
     }
 
     public void write(byte[] bytes) throws IOException {
-        if (process == null) {
+        if (!isOpen()) {
             throw new IllegalStateException("Process must be initialized (hint: call open() before write())");
         }
         processStandardInput.write(bytes);
     }
 
+    public boolean isOpen() {
+        return process != null;
```

Comment:
A call to [Process.isAlive()](https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#isAlive--) could be also relevant, as the documentation states:

> Returns:
true if the subprocess represented by this Process object has not yet terminated.

So if the process has died for some reason between writes, that is the method that would catch it. That implies changing a bit the exception message or showing 2 different messages depending on the situation.